### PR TITLE
Adding Fish Storage Struct

### PR DESCRIPTION
**Pull request by tkapa**
#12

Adds `fish_storage.rs` 

Attaches a FishStorage to the player entity.

Allows us to use `FishStorage::update_storage` to update a storage.

### CODE REVIEW

File: src/fish_storage.rs
Comment (SamBWagner) [inline src/fish_storage.rs:1]: I'm not conviced by fish storage. I think generic storage would be more pog.

**Review by SamBWagner — CHANGES_REQUESTED** [review]

File: src/fish_storage.rs
Comment (SamBWagner) [reply src/fish_storage.rs:1]: actually. I don't think it matters for nwo

Comment (tkapa) [reply src/fish_storage.rs:1]: How so?

Do you mean that we should just rename `FishStorage` to `Storage`?

Comment (SamBWagner) [reply src/fish_storage.rs:1]: I think the solution architecture could be more in line with having individual store components inside a larger `storage` component/entity
meaning you could compose any type of storage. But it's too much yours is more pog

**Review by SamBWagner — APPROVED** [review]

File: src/fish_storage.rs
Comment (tkapa) [reply src/fish_storage.rs:1]: For now, I think we only care about tracking the capacity of the storage, and the max value of the Storage

I think there's an argument for renaming this to `storage` instead and expanding upon your idea later if applicable.

**Pull request merged**